aptos-move/framework/supra-framework/sources/multisig_voting.spec.move

```diff
@@ -0,0 +1,427 @@
+        ensures (simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key) && len(next_execution_hash) != 0) ==>
+            simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key) == std::bcs::serialize(true);
+        ensures (simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key) &&
+            (len(next_execution_hash) == 0 && !is_multi_step)) ==>
+            simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key) == std::bcs::serialize(true);
+
+        let multi_step_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        aborts_if simple_map::spec_contains_key(proposal.metadata, multi_step_key) &&
+            !from_bcs::deserializable<bool>(simple_map::spec_get(proposal.metadata, multi_step_key));
+        let is_multi_step = simple_map::spec_contains_key(proposal.metadata, multi_step_key) &&
+            from_bcs::deserialize(simple_map::spec_get(proposal.metadata, multi_step_key));
+        aborts_if !is_multi_step && len(next_execution_hash) != 0;
+
+        aborts_if len(next_execution_hash) == 0 && !exists<timestamp::CurrentTimeMicroseconds>(@supra_framework);
+        aborts_if len(next_execution_hash) == 0 && is_multi_step && !simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key);
+        // property 4: For single-step proposals, it ensures that the next_execution_hash parameter is empty and resolves the proposal.
+        /// [high-level-req-4]
+        ensures len(next_execution_hash) == 0 ==> post_proposal.resolution_time_secs == timestamp::spec_now_seconds();
+        ensures len(next_execution_hash) == 0 ==> post_proposal.is_resolved == true;
+        ensures (len(next_execution_hash) == 0 && is_multi_step) ==> simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key) == std::bcs::serialize(false);
+        // property 4: For multi-step proposals, it ensures that the next_execution_hash parameter contains the hash of the next step.
+        ensures len(next_execution_hash) != 0 ==> post_proposal.execution_hash == next_execution_hash;
+    }
+
+    spec next_proposal_id {
+        aborts_if !exists<VotingForum<ProposalType>>(voting_forum_address);
+        ensures result == global<VotingForum<ProposalType>>(voting_forum_address).next_proposal_id;
+    }
+
+    spec get_proposer {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.proposer;
+    }
+
+    spec is_voting_closed {
+        use supra_framework::chain_status;
+        // Ensures existence of Timestamp
+        requires chain_status::is_operating();
+        //TODO: Remove pragma aborts_if_is_partial;
+        pragma aborts_if_is_partial = true;
+        include AbortsIfNotContainProposalID<ProposalType>;
```

Comment:
There is no `ensures` here, it seems commented out, why?

Reply:
Fixed in [1c5eacc](https://github.com/Entropy-Foundation/aptos-core/pull/63/commits/1c5eacc8283011e89d2ea0cd86e196e0ea79be7c)